src/Commands/Workflow/WaitCommand.php

```diff
@@ -175,69 +182,170 @@ protected function waitForWorkflow(
-    //             throw new TerminusException(
-    //                 'Exceeded maximum wait time of {max} seconds.',
-    //                 ['max' => $maxWaitInSeconds]
-    //             );
-    //         }
-    //         $this->log()->notice('Waiting for workflow {id} to complete.', ['id' => $wfl->id,]);
-    //         sleep($this->getConfig()->get('refresh_workflow_delay', 30));
-    //         $wfl->fetch();
-    //     }
-    //     $this->log()->notice('Workflow {id} has completed with status {status}.', [
-    //         'id' => $wfl->id,
-    //         'status' => $wfl->get('status'),
-    //     ]);
-    // }
+    protected function waitForCommit(
+        $startTime,
+        $site,
+        $env_name,
+        $target_commit,
+        $maxWaitInSeconds = 180
+    ) {
+        $current_time = time();
+        if ($maxWaitInSeconds > 0) {
+            $end_time = $current_time + $maxWaitInSeconds;
+        } else {
+            $end_time = 0;
+        }
+
+        // Validate commit SHA format
+        if (!preg_match('/^[0-9a-f]{40}$/', $target_commit)) {
+            throw new TerminusException(
+                'Commit {commit} is not a valid commit SHA.',
+                ['commit' => $target_commit]
+            );
+        }
+
+        $this->log()->notice('Waiting for workflow with commit {commit} on environment {env}.', [
+            'commit' => $target_commit,
+            'env' => $env_name
+        ]);
+        
+
+        $workflow = null;
+        $retry_count = 0;
+        $max_retries = 10;
+
+        do {
+            $current_time = time();
+            
+            // Check timeout
+            if ($end_time > 0 && $current_time >= $end_time) {
+                throw new TerminusException(
+                    'Workflow with commit {commit} timed out after {timeout} seconds.',
+                    ['commit' => $target_commit, 'timeout' => $maxWaitInSeconds]
+                );
+            }
+
+            // Fetch workflow logs using the logs/workflows endpoint
+            $response = $this->request()->request("sites/{$site->id}/logs/workflows");
+            $workflow_logs = $response['data'] ?? [];
+
+            $this->log()->debug('Found {count} total workflow logs', ['count' => count($workflow_logs)]);
+
+            // Filter for the target environment and commit
+            $matching_workflows = [];
+            
+            foreach ($workflow_logs as $log) {
+                // Check if this workflow is for the target environment
+                if (isset($log->workflow->environment) && $log->workflow->environment === $env_name) {
+                    // Check if this workflow has the target commit
+                    if (isset($log->workflow->target_commit) && $log->workflow->target_commit === $target_commit) {
+                        // Check if workflow started after our start time
+                        if (isset($log->workflow->started_at) && $log->workflow->started_at >= $startTime) {
```

Comment:
Now that we have the commit, maybe we should ignore the start time? I don't know how helpful it is, and folks often use it wrong, causing problems.

Reply:
Yeah I don't disagree.  We should probably solicit more opinions.

It does seem weird to care when the workflow started if what you're asking for is a workflow that deployed a specific commit hash.

Reply:
Fine, merge as is.

Follow on work: allow workflows that are still running / end time is after start time.